lib/datasets/dataset.rb

```diff
@@ -33,9 +33,23 @@ def cache_path
       @cache_path ||= CachePath.new(@metadata.id)
     end
 
-    def download(output_path, url, &block)
-      downloader = Downloader.new(url)
-      downloader.download(output_path, &block)
+    def download(output_path, url, *fallback_urls, &block)
+      urls = [url] + fallback_urls
+      urls.each_with_index do |url, idx|
+        downloader = Downloader.new(url)
+        downloader.download(output_path, &block)
+        return
+      rescue Net::HTTPClientException => error
+        if urls.last != url
+          failed_url = "<#{url}>"
+          fallback_url = "<#{urls[idx + 1]}>"
+          message = "#{error.response.code} #{error.response.message}: " +
+                    "fallback: #{failed_url} -> #{fallback_url}"
```

Comment:
`XXX_url` should not have `<` and `>` because `<` and `>` aren't URL:

```suggestion
          failed_url = url
          fallback_url = urls[idx + 1]
          message = "#{error.response.code} #{error.response.message}: " +
                    "fallback: <#{failed_url}> -> <#{fallback_url}>"
```

Reply:
I agree, `XXX_url` should not have `<` and `>` because `<` and `>` aren't part of the URL.